File: benchmarks/tsb/bench_read_json_all_orients.ts
/**
 * Benchmark: readJson with all orient options (records, split, columns, index, values).
 * Outputs JSON: {"function": "read_json_all_orients", "mean_ms": ..., "iterations": ..., "total_ms": ...}
 */
import { DataFrame, readJson, toJson } from "../../src/index.ts";

const SIZE = 5_000;
const WARMUP = 3;
const ITERATIONS = 20;

const df = DataFrame.fromColumns({
  id: Array.from({ length: SIZE }, (_, i) => i),
  value: Array.from({ length: SIZE }, (_, i) => i * 1.1),
  label: Array.from({ length: SIZE }, (_, i) => `cat_${i % 10}`),
});

const recordsJson = toJson(df, { orient: "records" });
const splitJson = toJson(df, { orient: "split" });
const columnsJson = toJson(df, { orient: "columns" });
const valuesJson = toJson(df, { orient: "values" });
const indexJson = toJson(df, { orient: "index" });

for (let i = 0; i < WARMUP; i++) {
  readJson(recordsJson, { orient: "records" });
  readJson(splitJson, { orient: "split" });
  readJson(columnsJson, { orient: "columns" });
  readJson(valuesJson, { orient: "values" });
  readJson(indexJson, { orient: "index" });
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  readJson(recordsJson, { orient: "records" });
  readJson(splitJson, { orient: "split" });
  readJson(columnsJson, { orient: "columns" });
  readJson(valuesJson, { orient: "values" });
  readJson(indexJson, { orient: "index" });
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "read_json_all_orients",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_read_json_all_orients.py
"""Benchmark: pd.read_json with all orient options (records, split, columns, index, values)."""
import json, time
import numpy as np
import pandas as pd

SIZE = 5_000
WARMUP = 3
ITERATIONS = 20

ids = list(range(SIZE))
values = [i * 1.1 for i in range(SIZE)]
labels = [f"cat_{i % 10}" for i in range(SIZE)]
df = pd.DataFrame({"id": ids, "value": values, "label": labels})

records_json = df.to_json(orient="records")
split_json = df.to_json(orient="split")
columns_json = df.to_json(orient="columns")
values_json = df.to_json(orient="values")
index_json = df.to_json(orient="index")

for _ in range(WARMUP):
    pd.read_json(records_json, orient="records")
    pd.read_json(split_json, orient="split")
    pd.read_json(columns_json, orient="columns")
    pd.read_json(values_json, orient="values")
    pd.read_json(index_json, orient="index")

start = time.perf_counter()
for _ in range(ITERATIONS):
    pd.read_json(records_json, orient="records")
    pd.read_json(split_json, orient="split")
    pd.read_json(columns_json, orient="columns")
    pd.read_json(values_json, orient="values")
    pd.read_json(index_json, orient="index")
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "read_json_all_orients",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
